crates/pipeline-manager/src/compiler.rs

```diff
@@ -585,12 +607,24 @@ inherits = "release"
                     "Program {} does not have a local artifact despite being in the {:?} state. Removing binary references to the program and re-queuing it for compilation.",
                     pipeline.id, pipeline.program_status
                 );
-                db.transit_program_status_to_pending(
-                    tenant_id,
-                    pipeline.id,
-                    pipeline.program_version,
-                )
-                .await?;
+                match db
+                    .transit_program_status_to_pending(
+                        tenant_id,
+                        pipeline.id,
+                        pipeline.program_version,
+                    )
+                    .await
+                {
+                    Ok(()) => {}
+                    Err(e) => match e {
+                        DBError::UnknownPipeline { .. } => {
+                            info!("Program {} cannot be re-queued because it no longer exists (tenant: {tenant_id})", pipeline.id);
```

Comment:
```suggestion
                            debug!("Program {} cannot be re-queued because it no longer exists (tenant: {tenant_id})", pipeline.id);
```